sdk/android/skin/src/main/java/com/ooyala/android/skin/OoyalaSkinLayout.java

```diff
@@ -94,45 +101,71 @@ public int getViewHeight() {
   }
 
   public boolean isFullscreen() {
-    return isFullscreen;
+    return fullscreen;
   }
 
 
   /**
    * Show/Hide system ui (notification and navigation bar) depending if layout is in fullscreen
    */
-  public void toggleSystemUI() {
-    if(isFullscreen()) {
-      setSystemUiVisibility(
-              View.SYSTEM_UI_FLAG_LAYOUT_STABLE
-                      | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
-                      | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
-                      | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION // hide nav bar
-                      | View.SYSTEM_UI_FLAG_FULLSCREEN // hide status bar
-                      | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
+  public void toggleSystemUI(boolean fullscreen) {
+    if(fullscreen) {
+      if (SDK_INT >= KITKAT) {
+        setSystemUiVisibility(
+                View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+                | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
+                | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+                | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION     // hide nav bar
+                | View.SYSTEM_UI_FLAG_FULLSCREEN          // hide status bar
+                | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);  // toggle system UI visibility automatically
+      } else {
+        setSystemUiVisibility(
+                View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+                | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
+                | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+                | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION     // hide nav bar
+                | View.SYSTEM_UI_FLAG_FULLSCREEN);        // hide status bar
+      }
     } else {
       setSystemUiVisibility(
               View.SYSTEM_UI_FLAG_LAYOUT_STABLE);
     }
   }
 
   /**
-   * Stretch OoyalaSkinLayout to dimensions of parent layout.
+   * Stretch OoyalaSkinLayout to dimensions of the display window.
    * Handle system UI visibility.
    */
   void setFullscreen(boolean fullscreen) {
```

Comment:
yes it can and it's valid. I think you mean is `windowManager` used before it is even created? The answer should be no, but i am not a 100% sure.

I am initalizing `windowManager` whenever you create an instance of `OoyalaSkinLayout`, so I don't see it failing.

Anyway, I will be extra safe and add a null checker in case `windowManager` is not initialized for some strange reason.